pkgs/tools/virtualization/google-guest-configs/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper
+, ipcalc, iproute2, util-linux, coreutils, ethtool, gnugrep, gnused, nvme-cli }:
+
+stdenv.mkDerivation rec {
+  pname = "google-guest-configs";
+  version = "20211116.00";
+
+  src = fetchFromGitHub {
+    owner = "GoogleCloudPlatform";
+    repo = "guest-configs";
+    rev = version;
+    sha256 = "sha256-0SRu6p/DsHNNI20mkXJitt/Ee5S2ooiy5hNmD+ndecM=";
+  };
+
+  binDeps = lib.makeBinPath [ coreutils util-linux gnugrep gnused ethtool ipcalc iproute2 ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/{bin,etc,lib}
+    cp -r src/etc/{modprobe.d,sysctl.d} $out/etc
+    cp -r src/lib/udev $out/lib
+    cp -r src/sbin/* $out/bin
+    cp -r src/usr/bin/* $out/bin
+
+    for i in "$out/lib/udev/rules.d"/*.rules; do
+      sed -i \
+        -e "s,\"google_nvme_id,\"$out/lib/udev/google_nvme_id,g" \
+        -e "s,/bin/sh,${stdenv.shell},g" \
+        -e "s,/bin/umount,${util-linux}/bin/umount,g" \
+        -e "s,/usr/bin/logger,${util-linux}/bin/logger,g" \
+        "$i"
+    done
+
+    sed -i \
+      -e 's,/bin/ipcalc,ipcalc,' \
+      -e "/^PATH=/d" \
+      $out/bin/google-dhclient-script
+    sed -i \
+      -e 's,/usr/sbin/nvme,${nvme-cli}/bin/nvme,' \
+      $out/lib/udev/google_nvme_id
```

Comment:
This will silently fail to substitute whenever those don't apply anymore.

Can this be changed to use a patch replacing the strings with @placeholders@, and `substituteAll`?

See https://github.com/NixOS/nixpkgs/commit/6b1937818e7f18ee7eac034cb9ccc84db0087dfc for an example.

Reply:
Good point, fixed. I use `substituteInPlace` call instead because we want `@out@` to point to the package's `$out`.